core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -330,10 +332,21 @@ class KafkaServer(
         groupCoordinator = GroupCoordinator(config, replicaManager, Time.SYSTEM, metrics)
         groupCoordinator.startup(() => zkClient.getTopicPartitionCount(Topic.GROUP_METADATA_TOPIC_NAME).getOrElse(config.offsetsTopicPartitions))
 
+        /* create producer ids manager */
+        val producerIdManager = if (config.interBrokerProtocolVersion.isAllocateProducerIdsSupported) {
+          ProducerIdGenerator(
```

Comment:
It seems a bit weird that both function calls both have the same name. Can't we just create the subclass we want directly here, for clarity? One is ZkWhatever, the other is ForwardingWhatever, etc.